### DISCUSSION TITLE
How to set slug field

### DISCUSSION THREAD

**Discussion by raakkan** [filamentphp/filament#451] [Help] [answered]
category model has name field and slug field. this slug generate from name field. please help anyone.

**Comment by loranger** [comment, marked as answer]
I had the same problem.

I dug in the discussion and was redirected to the [hooks part](https://filamentadmin.com/docs/resources/#pages-customization-hooks) of the documentation. But in my opinion, sometimes filament documentation is too light to be really helpful.

In your case, you have to override some methods.
I had a Product Model with a slug attribute (generated from the name attribute) for which I needed to create a CRUD, so I created a resource (`php artisan make:filament-resource Product`).

Inside my `app/Filament/Resources/ProductResource/Pages/CreatePorudct.php` file, I added the following method:

```php
...

use Illuminate\Support\Str;

class CreateProduct extends CreateRecord
{

...

    public function updatedRecordName($value)
    {
        $slug = Str::slug($value);
        $this->record['sku'] = $slug;
    }
```

I didn't need to make this sku attribute editable, I only needed to generate this slug on creation.
But if you need this edition, you'll have to add this method in EditYourModel php page.

I hope it would help !
(This method is not documented, I would really love to find such a tip & trick in the documentation)
